src/json_decode.mli

```diff
@@ -208,11 +208,26 @@ and a value that is successfully decoded with the given decoder, [Error of strin
   Js.log \@\@ Js.Json.parseExn {| { "x": 23, "y": "b" } |} |> Decode.(field "y" int)
   (* prints [Error ...] *)
   Js.log \@\@ Js.Json.parseExn "123" |> Decode.(field "x" int)
-  (* prints [Ok None] *)
+  (* prints [Error ...] *)
```

Comment:
Ah, good catch! I also need to rewrite all these since the API now uses exceptions instead of returning a `result` type.